src/apps/common/submodules/portListing/portListing.js

```diff
@@ -780,108 +867,115 @@ define(function(require) {
+		 * of transitions only.
+		 */
+		portListingFormatEntriesToTimeline: function(entries) {
 			var self = this,
-				comments = entries.filter(function(entry) { return entry.hasOwnProperty('author'); }),
-				statuses = entries.filter(function(entry) { return entry.hasOwnProperty('transition') && entry.transition.new !== 'unconfirmed'; }),
-				timeline = [];
-
-			_.each(comments, function(comment) {
-				timeline.push({
-					timestamp: moment(monster.util.gregorianToDate(comment.timestamp)).valueOf(),
-					title: comment.author,
-					content: comment.content
-				});
-			});
-
-			_.each(statuses, function(status, idx) {
-				timeline.push({
-					timestamp: moment(monster.util.gregorianToDate(status.timestamp)).valueOf(),
-					// do not show the `Action Required` text if the status is rejected by not the most recent
-					title: self.i18n.active().portListing.misc[(statuses.length - 1 !== idx) && status.transition.new === 'rejected' ? 'alternateStatus' : 'status'][status.transition.new],
-					content: status.reason,
-					status: status.transition.new,
-					// show fix button if the last status is rejected, this is the cause of filter() & double each()
-					needFix: (statuses.length - 1 === idx) && status.transition.new === 'rejected'
-				});
-			});
-
-			return _.sortBy(timeline, 'timestamp');
-		},
+				isLastAndRejected = function(transition, idx, transitions) {
+					return transitions.length - 1 === idx && transition.transition.new === 'rejected';
+				};
 
-		portListingGenerateTimeline: function(entries) {
-			var self = this,
-				formattedEntries = self.portListingFormatToTimeline(entries),
-				entriesByDays = _.groupBy(formattedEntries, function(entry) {
-					// group entries by calendar days
+			return _
+				.chain([
+					_.chain(entries)
+						.filter('author')
+						.map(function(comment) {
+							return {
+								content: comment.content,
+								timestamp: moment(monster.util.gregorianToDate(comment.timestamp)).valueOf(),
+								title: comment.author
+							};
+						})
+						.value(),
+					_.chain(entries)
+						.reject(function(entry) {
+							return _.has(entry, 'author') || _.get(entry, 'transition.new', 'unconfirmed') === 'unconfirmed';
+						})
+						.map(function(transition, idx, array) {
+							return {
+								content: _.get(transition, 'reason', undefined),
+								// show fix button if the last transition's status is "rejected"
+								needFix: isLastAndRejected(transition, idx, array),
+								status: transition.transition.new,
+								timestamp: moment(monster.util.gregorianToDate(transition.timestamp)).valueOf(),
+								// only show the "Action Required" text if the last transition's status is "rejected"
+								title: self.i18n.active().portListing.misc[isLastAndRejected(transition, idx, array) ? 'alternateStatus' : 'status'][transition.transition.new]
+							};
+						})
+						.value()
+				])
+				.reduce(function(acc, item) {
```

Comment:
I think this `reduce(...)` could be replaced by [`flatten()`](https://lodash.com/docs/4.17.11#flatten)

Reply:
Nice, I will improve this in my next PR (UI-3259).